doc/indexing.rst

```diff
@@ -357,6 +357,13 @@ These methods may and also be applied to ``Dataset`` objects
   ``isel_points`` and ``sel_points`` are now deprecated.
   See :ref:`more_advanced_indexing` for their alternative.
 
+.. note::
+
+  If an indexer is a :py:meth:`~xarray.DataArray`, its coordinates should not
+  conflict with the selected subpart of the target array.
```

Comment:
We should probably note the exception for explicitly indexed dimensions with `.loc`/`.sel`?